compendium/ServiceComponent/CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+# sources and headers
+add_subdirectory(src)
+
+set(_version "1.0.0")
+project(ServiceComponent VERSION ${_version})
+
+if(NOT ${PROJECT_NAME}_TARGET)
+  set(${PROJECT_NAME}_TARGET usServiceComponent)
+endif()
+set(PROJECT_TARGET ${${PROJECT_NAME}_TARGET})
+
+# link libraries for the ServiceComponent lib
+set(_link_libraries )
+if(UNIX)
+  list(APPEND _link_libraries dl)
+endif()
+
+if(CMAKE_THREAD_LIBS_INIT)
+  list(APPEND _link_libraries ${CMAKE_THREAD_LIBS_INIT})
+endif()
+
+configure_file(${CppMicroServices_SOURCE_DIR}/cmake/Export.h.in
+  ${CMAKE_CURRENT_BINARY_DIR}/include/cppmicroservices/servicecomponent/ServiceComponentExport.h
+  )
+
+add_library(${PROJECT_TARGET} $<TARGET_OBJECTS:ServiceComponentObjs> ${CMAKE_CURRENT_BINARY_DIR}/include/cppmicroservices/servicecomponent/ServiceComponentExport.h)
+
+target_include_directories(usServiceComponent
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+    $<BUILD_INTERFACE:${CppMicroServices_BINARY_DIR}/include>
+    $<INSTALL_INTERFACE:${HEADER_INSTALL_DIR}>
+  )
+
+include_directories(${CppMicroServices_SOURCE_DIR}/framework/include
```

Comment:
I don't believe both ```target_include_directories``` and ```include_directories``` are needed. They do the same thing just at different scopes (i.e. target vs. directory scope). 

I suggest removing ```include_directories```

Reply:
Done.